expression/builtin_other.go

```diff
@@ -52,6 +61,252 @@ var (
 	_ builtinFunc = &builtinGetParamStringSig{}
 )
 
+type inFunctionClass struct {
+	baseFunctionClass
+}
+
+func (c *inFunctionClass) getFunction(ctx context.Context, args []Expression) (sig builtinFunc, err error) {
+	if err := c.verifyArgs(args); err != nil {
+		return nil, errors.Trace(err)
+	}
+	argTps := make([]types.EvalType, len(args))
+	for i := range args {
+		argTps[i] = args[i].GetType().EvalType()
+	}
+	bf := newBaseBuiltinFuncWithTp(ctx, args, types.ETInt, argTps...)
+	bf.tp.Flen = 1
+	switch args[0].GetType().EvalType() {
+	case types.ETInt:
+		sig = &builtinInIntSig{baseBuiltinFunc: bf}
+		sig.setPbCode(tipb.ScalarFuncSig_InInt)
+	case types.ETString:
+		sig = &builtinInStringSig{baseBuiltinFunc: bf}
+		sig.setPbCode(tipb.ScalarFuncSig_InString)
+	case types.ETReal:
+		sig = &builtinInRealSig{baseBuiltinFunc: bf}
+		sig.setPbCode(tipb.ScalarFuncSig_InReal)
+	case types.ETDecimal:
+		sig = &builtinInDecimalSig{baseBuiltinFunc: bf}
+		sig.setPbCode(tipb.ScalarFuncSig_InDecimal)
+	case types.ETDatetime, types.ETTimestamp:
+		sig = &builtinInTimeSig{baseBuiltinFunc: bf}
+		sig.setPbCode(tipb.ScalarFuncSig_InTime)
+	case types.ETDuration:
+		sig = &builtinInDurationSig{baseBuiltinFunc: bf}
+		sig.setPbCode(tipb.ScalarFuncSig_InDuration)
+	case types.ETJson:
+		sig = &builtinInJSONSig{baseBuiltinFunc: bf}
+		sig.setPbCode(tipb.ScalarFuncSig_InJson)
+	}
+	return sig, nil
+}
+
+type builtinInIntSig struct {
+	baseBuiltinFunc
+}
+
+func (b *builtinInIntSig) evalInt(row types.Row) (int64, bool, error) {
```

Comment:
add a comment to the mysql function link: [https://dev.mysql.com/doc/refman/5.7/en/comparison-operators.html#function_in](https://dev.mysql.com/doc/refman/5.7/en/comparison-operators.html#function_in)

Reply:
added at the head of each sig.